conftest.py

```diff
@@ -0,0 +1,8 @@
+# NOTE: This is required for the MyPy test suite machinery to pick up
```

Comment:
Does this file have to be at the root? Or can we move this to `thinc/tests/conftest.py`?

Reply:
Yeah, sadly it has to be here in order to register the mypy test runner; if it's in a subfolder pytest complains. 

I'm thinking we could refactor this away and use our own mypy test runner that's less complicated and hardcoded. I skipped it for now to keep things simple until we verify it's useful

Reply:
I think I found a decent solution. I inlined the test runner bits we need from mypy and removed the various ENV and conftest.py hacks from before